File: Challenge#02/Challenge_02.py
def spy_language(operations: str) -> str:
    '''
    Calcula un número a base de operaciones utilizando caracteres especiales
    # = suma 1, @ = resta 1, * = cuadrado, & = fin
    '''
    number = 0
    out = ""

    for operation in operations:
        if operation == "#":
            number += 1
        elif operation == "@":
            number -= 1
        elif operation == "*":
            number *= number
        elif operation == "&":
            out += str(number)
    return out


print(spy_language("&###@&*&###@@##@##&######@@#####@#@#@#@##@@@@@@@@@@@@@@@*&&@@@@@@@@@####@@@@@@@@@#########&#&##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@&"))

File: Challenge#02/Challenge_02.js
let code = "&###@&*&###@@##@##&######@@#####@#@#@#@##@@@@@@@@@@@@@@@*&&@@@@@@@@@####@@@@@@@@@#########&#&##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@&"

function spy_code(code) {
    let number = 0;
    let out = "";

    for (operation of code) {
        if (operation == "#") {
            number += 1;
        } else if (operation == "@") {
            number -= 1;
        } else if (operation == "*") {
            number *= number
        } else if (operation == "&") {
            out += number.toString();
        }
    }
    
    return out;
}

console.log(spy_code(code));
